packages/orchestration/src/exos/stakingAccountKit.js

```diff
@@ -261,9 +372,51 @@ export const prepareStakingAccountKit = (baggage, makeRecorderKit, zcf) => {
           if (!balance) throw Fail`Result lacked balance key: ${result}`;
           return harden(toChainAmount(balance));
         },
+
+        withdrawRewards() {
+          throw assert.error('Not implemented');
+        },
+
+        /**
+         * @param {Delegation[]} delegations
+         */
+        async undelegate(delegations) {
+          trace('undelegate', delegations);
+          const { helper } = this.facets;
+          const { chainAddress, bondDenom } = this.state;
+
+          const result = await E(helper.owned()).executeEncodedTx(
+            delegations.map(d =>
+              toAnyJSON(
+                MsgUndelegate.toProtoMsg({
+                  delegatorAddress: chainAddress.address,
+                  validatorAddress: d.validatorAddress,
+                  amount: { denom: bondDenom, amount: d.shares },
+                }),
+              ),
+            ),
+          );
+
+          const response = tryDecodeResponse(
+            result,
+            MsgUndelegateResponse.fromProtoMsg,
+          );
+          trace('undelegate response', response);
+          // TODO? return completionTime?
+        },
       },
     },
   );
+
+  /** check holder facet against StakingAccountActions interface. FIXME: unused */
+  const typeCheck = () => {
+    /** @import {Guarded, Methods} from '@endo/exo'; */
+    /** @type {any} */
+    const arg = null;
+    /** @satisfies { StakingAccountActions } */
+    const kit = makeStakingAccountKit(arg, arg, arg, arg, arg).holder;
+  };
```

Comment:
@turadg do you have any better idea than to keep this and silence the unused warnings?